### PR TITLE
Add tutorial for distributed training and batch prediction with Kubernetes

### PR DESCRIPTION
**Pull request by merlintang**
## What is proposed in this PR ?
The readme tutorial to train XGBoost job on Kubernetes cluster is added. 

## How to test this PR ? 
Just readme, and related tutorial is linked.

### CODE REVIEW

**Comment by merlintang** [comment]
@CodingCat and @hcho3 can you review this ?

File: doc/tutorials/kubernetes.rst [hunk 1]
```diff
@@ -0,0 +1,40 @@
+###############################
+Distributed XGBoost on Kubernetes cluster
+###############################
+
+Distributed XGBoost includes model training and batch prediction. Beside the XGBoost over Spark, now, XGBoost job can run in Kubernetes cluster natively.
```
Comment (yinlou) [inline doc/tutorials/kubernetes.rst]: Beside -> In addition to

File: doc/tutorials/kubernetes.rst [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  
+  - XGBoost Operator is designed to manage the XGBoost Job like job scheduling, monitoring, pods and services recovering etc. Thus, you can follow the `installment <https://github.com/kubeflow/xgboost-operator#installing-xgboost-operator>`_ guide to make XGBoost Job controller work at first.  
+
+- Step2: Write your application and submit to XGBoost job controller
+
+  - you can follow the `example <https://github.com/kubeflow/xgboost-operator/tree/master/config/samples/xgboost-dist>`_ to develop your own XGBoost application. 
+
+  - Data reader/writer: you need to have your data source reader and writer based on the requirement. For example, if your data is stored in one Hive Table, you have to build your own code to read/write Hive table based on the ID of worker. 
+
+  - Model persistence: in this example, model is stored in the OSS storage. If you want to store your model into Amazon S3, Google NFS or other storage. You also need to specific the model reader and writer based on the requirement of storage system.
```
Comment (yinlou) [inline doc/tutorials/kubernetes.rst]: specific -> specify

File: doc/tutorials/kubernetes.rst [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+  - you can follow the `example <https://github.com/kubeflow/xgboost-operator/tree/master/config/samples/xgboost-dist>`_ to develop your own XGBoost application. 
+
+  - Data reader/writer: you need to have your data source reader and writer based on the requirement. For example, if your data is stored in one Hive Table, you have to build your own code to read/write Hive table based on the ID of worker. 
+
+  - Model persistence: in this example, model is stored in the OSS storage. If you want to store your model into Amazon S3, Google NFS or other storage. You also need to specific the model reader and writer based on the requirement of storage system.  
+
+- Step3: Configure your Yaml file 
+
+  - Yaml file is used to configure the computation resource and environment for your XGBoost job to run. Therefore, you can configure the Yaml file to specific your job runtime like the number of worker and master. The template `Yaml <https://github.com/kubeflow/xgboost-operator/blob/master/config/samples/xgboost-dist/xgboostjob_v1alpha1_iris_train.yaml>`_ is provided for you to refer.
```
Comment (yinlou) [inline doc/tutorials/kubernetes.rst]: specific -> speficy

File: doc/tutorials/kubernetes.rst [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+- Step4: Submit job 
+
+  - `Kubectl command <https://github.com/kubeflow/xgboost-operator#creating-a-xgboost-trainingprediction-job>`_ is used to submit a XGBoost job, and then you can monitor the job status on time.
```
Comment (yinlou) [inline doc/tutorials/kubernetes.rst]: remove "on time"

File: doc/tutorials/kubernetes.rst [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+- Step3: Configure your Yaml file 
+
+  - Yaml file is used to configure the computation resource and environment for your XGBoost job to run. Therefore, you can configure the Yaml file to specific your job runtime like the number of worker and master. The template `Yaml <https://github.com/kubeflow/xgboost-operator/blob/master/config/samples/xgboost-dist/xgboostjob_v1alpha1_iris_train.yaml>`_ is provided for you to refer.
+
+- Step4: Submit job 
+
+  - `Kubectl command <https://github.com/kubeflow/xgboost-operator#creating-a-xgboost-trainingprediction-job>`_ is used to submit a XGBoost job, and then you can monitor the job status on time. 
+
+**************
+Not Support
```
Comment (yinlou) [inline doc/tutorials/kubernetes.rst]: Not supported yet

**Comment by merlintang** [comment]
@yinlou  thanks for your comments, it is fixed.

**Comment by hcho3** [comment]
@merlintang I made minor edits for greater clarity. Can you review?

**Comment by hcho3** [comment]
Merging this for now. Feel free to open a new pull request to fix typos etc.

**Comment by merlintang** [comment]
thanks for Cho, I would update the typo later.

**Comment by hcho3** [comment]
@merlintang Here's is the official link: https://xgboost.readthedocs.io/en/latest/tutorials/kubernetes.html. Thank you for your contribution.

**Pull request merged**